### PR TITLE
recategorized speciality and misc

### PR DESCRIPTION
**Pull request by story645**
## PR Summary
Removed specialty and recategorized specialty and misc into:
* plotting utilities - adjust text, highlight text, broken axes, etc - not specific to any plot type, general tools
* plot types - plot types not specific to any domain - blume, windrose, venn, etc
* domain specific libraries - astropy, seaborn, mplfinance, etc
* documentation - libs for documenting mpl packages - sphinx gallery, @ianhi 's mpl recorder tool would be useful here too, 

doc could also be renamed to development and be a catch all for libraries for developing mpl packages like sphinx gallery, mpl test, mpl cookie cutter

### CODE REVIEW

**Comment by jklymak** [comment]
We can certainly jiggle the categories, but I feel some of these are too fine-scale. The page will be all headings with just a couple of entries in each and that will make it harder to read.  

"Declarative library" is *really* jargon-y.  

I think we don't need to have a difference between "Specialty plots" and "Domain-specific plots"

**Comment by jklymak** [comment]
The categories I used are just the existing ones.  I think if we are to recategorize it should probably be a more fulsome conversation with a bit of brain-storming.

**Review by jklymak — CHANGES_REQUESTED** [review]
I agree that the categories could be reworked.  However, I'll block because I think this needs discussion.  The categories as they are, are the same as the existing long-standing categories.

**Comment by story645** [comment]
I deleted speciality & pulled  everything out of miscellaneous because it wasn't clear to me which packages fell under which category, and instead I'm proposing what I think are categories with clearer distinctions. I will totally change the category for the declarative cause I agree with you that it's not worth having a whole category.

**Comment by jklymak** [comment]
... also right now I'm sorting alphabetical for the category names, but that is perhaps not ideal.   But that will require a bit of coding change to `build.py`

**Comment by jklymak** [comment]
PS, I'm fine w/ reorganizing, but its probably more than just a one-review change.

**Comment by story645** [comment]
Yeah, waiting on discussion is fine. I just figured this would be a "please out in pr that we can discuss" type situation

File: python/build.py [hunk 1]
```diff
@@ -18,15 +18,17 @@
...
-section_names = {'colormaps and styles': 'Colormaps and styles', 
-           'specialty plots': 'Specialty plots',
-           'gui applications': 'GUI applications', 
-           'miscellaneous': 'Miscellaneous', 
-           'backends': 'Rendering backends', 
+section_names = {'colormaps and styles': 'Colormaps and Styles', 
+           'plotting utilities': 'Plotting Utilities',
+           'plot types': 'Plot Types', 
+           'gui applications': 'GUI Applications', 
+           'backends': 'Rendering Backends',
```
Comment (jklymak) [inline python/build.py:25]: I am pretty sure we usually use sentence case instead of title case?

File: python/build.py [hunk 2]
```diff
@@ -18,15 +18,17 @@
...
+section_names = {'colormaps and styles': 'Colormaps and Styles', 
+           'plotting utilities': 'Plotting Utilities',
+           'plot types': 'Plot Types', 
+           'gui applications': 'GUI Applications', 
+           'backends': 'Rendering Backends', 
            'interactivity': 'Interactivity',
            'animations': 'Animations',
            'mapping': 'Mapping', 
-           'declarative libraries': 'Declarative Libraries'}
+           'domain specific libraries': 'Domain Specific Libraries',
```
Comment (jklymak) [inline python/build.py:29]:
```suggestion
           'domain specific libraries': 'Domain-Specific Libraries', 
```

**Review by jklymak — APPROVED** [review]
This org is probably fine.  Not sure I agree with the use of title case.  I think some of the libraries we have called "domain specific" are not terrible domain-specific.  If we add xarray and pandas, then we can maybe have a high-level analysis packages for seaborn and plotnine?

File: python/build.py (hunk 1 above)
Comment (story645) [reply python/build.py:25]: I thought sentence case read kinda weird, and these are titles?

**Comment by story645** [comment]
seaborn is stats domain specific, and plotnine is also kinda stats oriented since it's a ggplot clone? I'm using domain specific 'cause it's language we've used a lot in grant applications for these sorts of libraries, mostly 'cause they tend to incorporate domain specific semantics/assumptions into their APIs, but I'm also not opposed to "high level"

File: python/build.py (hunk 1 above)
Comment (jklymak) [reply python/build.py:25]: We should have a style for this. But I'll note that pandas and numpy both use sentence case below h1, and I think we should as well.

**Comment by jklymak** [comment]
"Statistics" is a tool used across every scientific domain.  Whereas astronomical projections really only apply to astronomy (for instance).

**Comment by story645** [comment]
Yeah, but statistics is also a specific discipline & seaborn and plotnine use the conventions of stats. but also, the distinction I really want to make is between libraries like seaborn and packages that only provide one type of plot like blume

**Comment by jklymak** [comment]
> Yeah, but statistics is also a specific discipline & seaborn and plotnine use the conventions of stats. 

I very strongly disagree that statistical analysis of data is "domain-specific" to statisticians.

**Comment by story645** [comment]
> I very strongly disagree that statistical analysis of data is "domain-specific" to statisticians.
The point I'm trying to make is that the seaborn API uses the conventions of statistics, not that statistics is only used by statisticians.

File: python/build.py (hunk 1 above)
Comment (anntzer) [reply python/build.py:25]: fwiw I Prefer Sentence Case As Title Case Is Just Weird.

**Comment by jklymak** [comment]
Lets merge this and fine-tune later.

**Pull request merged**